src/Workspaces/Core/Portable/Rename/SymbolicRenameLocations.ReferenceProcessing.cs

```diff
@@ -121,6 +128,18 @@ static bool IsConstructorForType(ISymbol possibleConstructor, ISymbol possibleTy
             }
         }
 
+        private static bool IsPropertyGeneratedFromPrimaryConstructorParameter(
+            ISymbol propertySymbol, ISymbol parameterSymbol, CancellationToken cancellationToken)
+        {
+            if (parameterSymbol is IParameterSymbol parameter)
+            {
+                var generatedPropertySymbol = parameter.GetAssociatedSynthesizedRecordProperty(cancellationToken);
+                return propertySymbol.Equals(generatedPropertySymbol);
```

Comment:
not sure if we need to sprinkle in .OriginalDefinition here in generic cases.  see test recommendation above.